symforce/codegen/type_description.py

```diff
@@ -0,0 +1,28 @@
+from symforce import typing as T
+from symforce.values.index_entry import IndexEntry
+
+
+class TypeDescription(object):
+    """
+    Description of a type. Contains all the data required to code-generate a type definition.
+    """
+
+    def __init__(self,
+                 typename: str,
+                 unformatted_typenames: T.List[str],
+                 full_typename: str,
+                 index: T.Dict[str, IndexEntry]):
+        """
+        Construct new TypeDescription.
+        Args:
+            TODO(gareth): Fill this out.
```

Comment:
Might as well delete this TODO, I don't have anything substantive to add here. Maybe add a comment on this object that it just replaces the dict that was previously built by `build_types_dict`

Reply:
done!